src/collector.rs

```diff
@@ -167,10 +167,42 @@ impl Collector for OperatingSystem {
             .as_ref()
             .map(|t| t.deref())
             .unwrap_or("(unknown version)");
+
+        #[cfg(target_os = "macos")]
+        return Ok(ReportEntry::Text(format!("{} ({} {})", macos_info_string(), os_type, os_release)));
```

Comment:
I'm not really happy about having to do it this way, but it didn't make sense to use an additional function just for a one line difference.